src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
         }
 
         public override async Task<IQueueEntry<T>> DequeueAsync(TimeSpan? timeout = null) {
+            await EnsureQueueCreatedAsync().AnyContext();
```

Comment:
I started this PR with it returning the queue client but moved away to have more code in the base implementation...

We could wrap `_queueClient` in a property that would throw a more meaningful exception? Even updating `EnsureQueueCreatedAsync()` to return the client wouldn't and adding locking in `DeleteQueueAsync()` wouldn't guarantee 100% it would still exist.